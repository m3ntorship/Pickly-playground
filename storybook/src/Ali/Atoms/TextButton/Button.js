import React from "react";
import PropTypes from "prop-types";

const Button = (props) => {
  const {
    textButtonDisabled,
    textButtonsize,
    textButtonvariant,
    textButtonchildren,
  } = props;

  return (
    <button
      disabled={textButtonDisabled}
      className={`capitalize rounded-full focus:outline-none disabled:opacity-25 ${
        textButtonvariant === "Primary"
          ? `bg-primary-shd1  text-white ${
              !textButtonDisabled && "hover:bg-primary"
            } focus:border-primary focus:border focus:ring-1 focus:ring-primary-shd1`
          : `border text-primary-shd1 border-primary-shd1 ${
              !textButtonDisabled && "hover:bg-grey-bg"
            } focus:ring-1 focus:ring-primary-shd1`
      } ${
        textButtonsize === "Large"
          ? "text-md font-bold px-xl py-3.5 "
          : textButtonsize === "Medium"
          ? "text-base font-medium px-l py-xs "
          : "text-sm font-medium px-m py-1.5 "
      }
      ${textButtonDisabled && "cursor-not-allowed"}`}
    >
      {textButtonchildren}
    </button>
  );
};

Button.propTypes = {
  textButtonsize: PropTypes.oneOf(["Small", "Medium", "Large"]),
  textButtonvariant: PropTypes.oneOf(["Primary", "Secondary"]),
  textButtonDisabled: PropTypes.bool,
  textButtonchildren: PropTypes.string,
};
export default Button;
