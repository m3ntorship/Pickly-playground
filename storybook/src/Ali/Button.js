import React from "react";
import PropTypes from "prop-types";

const Button = (props) => {
  const { disabled, size, variant, children } = props;
  return (
    <button
      disabled={disabled}
      className={`capitalize rounded-full font-bold focus:outline-none disabled:opacity-25 ${
        variant === "Primary"
          ? `bg-primary-shd1  text-white ${
              !disabled && "hover:bg-primary"
            } focus:border-primary focus:border focus:ring-1 focus:ring-primary-shd1`
          : `border text-primary-shd1 border-primary-shd1 ${
              !disabled && "hover:bg-grey-bg"
            } focus:ring-1 focus:ring-primary-shd1`
      } ${
        size === "Large"
          ? "text-md px-xl py-s "
          : size === "Medium"
          ? "text-base px-l py-xs "
          : "text-sm px-m py-xxs "
      }
      ${disabled ? "cursor-not-allowed" : ""}`}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  size: PropTypes.oneOf(["Small", "Medium", "Large"]),
  variant: PropTypes.oneOf(["Primary", "Secondary"]),
  disabled: PropTypes.bool,
  children: PropTypes.string,
};
export default Button;
