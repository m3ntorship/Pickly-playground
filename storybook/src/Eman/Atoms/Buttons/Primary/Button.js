import React from "react";
import PropTypes from "prop-types";

export const BUTTON_OPTIONS = {
  SIZE: {
    SMALL: "small",
    MEDIUM: "medium",
    BIG: "big",
  },
  TYPE: {
    PRIMARY: "primary",
    SECONDARY: "secondary",
  },
  DISABLED: {
    ON: "on",
    OFF: "off",
  },
};

export const Button = ({ type, content, size, disabled }) => {
  return (
    <button
      className={`
  ${
    type === BUTTON_OPTIONS.TYPE.PRIMARY
      ? "rounded-full  bg-primary  text-white hover:bg-primary-shd8 focus:outline-none focus:border-2 focus:border-primary-shd9 focus:border-opacity-100 disabled:opacity-5"
      : "rounded-full  text-primary  border-2 border-primary"
  },
        ${size === BUTTON_OPTIONS.SIZE.BIG && "px-xl py-s font-bold text-md"},
        ${
          size === BUTTON_OPTIONS.SIZE.MEDIUM &&
          "px-l py-xs font-semibold  text-base"
        },
        ${
          size === BUTTON_OPTIONS.SIZE.SMALL &&
          "px-m py-xxs font-semibold text-sm"
        },
        ${disabled === BUTTON_OPTIONS.DISABLED.ON && "opacity-25"}
        `}
    >
      {content}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.string,
  content: PropTypes.string,
  size: PropTypes.string,
  disabled: PropTypes.string,
};
