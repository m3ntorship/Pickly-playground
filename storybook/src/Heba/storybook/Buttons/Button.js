import React from "react";
import cn from "classnames";

export const BUTTON_OPTIONS = {
  SIZE: {
    SMALL: "small",
    MEDIUM: "medium",
    BIG: "big",
  },
};

export const Button = ({ primary, size }) => {
  return (
    <div className="flex justify-start items-center ">
      <button
        type="button"
        className={cn(
          "flex  items-center justify-center",
          primary
            ? "text-white bg-primary rounded-full focus:ring-2 focus:ring-primary-shd9 focus:outline-none disabled:opacity-25 mr -4"
            : " text-primary bg-white  border border-primary rounded-full  focus:ring-2 focus:ring-primary-shd9 focus:outline-none mr-4 disabled:opacity-25 fill-current",
          { "py-3 px-3 ": size === BUTTON_OPTIONS.SIZE.BIG },
          {
            "py-2 px-2": size === BUTTON_OPTIONS.SIZE.MEDIUM,
          },

          {
            "py-2 px-2": size === BUTTON_OPTIONS.SIZE.SMALL,
          }
        )}
        disabled
      >
        <svg
          className={cn(
            { "w-8 h-auto ": size === BUTTON_OPTIONS.SIZE.BIG },
            {
              "w-auto h-6": size === BUTTON_OPTIONS.SIZE.MEDIUM,
            },

            {
              "w-3 h-auto": size === BUTTON_OPTIONS.SIZE.SMALL,
            }
          )}
          viewBox="0 0 32 32"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.724 12.3906L16 18.1146L10.276 12.3906L8.39062 14.276L16 21.8853L23.6093 14.276L21.724 12.3906Z"
            className={cn(
              primary ? "text-white fill-current" : "text-primary fill-current"
            )}
          />
        </svg>
      </button>
    </div>
  );
};
