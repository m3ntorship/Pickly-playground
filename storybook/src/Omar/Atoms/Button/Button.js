import React from "react";
import PropTypes from "prop-types";

export const Button = (props) => {
  const {
    disabled = false,
    size = "sm",
    variant = "primary",
    content,
    onClick,
  } = props;

  return (
    <button
      data-testid="btn-case"
      disabled={disabled}
      onClick={onClick}
      className={`rounded-full border disabled:opacity-30 flex items-center ${
        variant === "primary"
          ? "bg-primary text-white hover:bg-primary-shd9 focus:outline-none focus:border-primary-shd9"
          : "bg-white text-primary border-primary border-solid hover:bg-grey-bg2 focus:border-2 focus:outline-none"
      } ${
        size === "lg"
          ? "text-md py-3.5 px-xl"
          : size === "sm"
          ? "text-sm py-1.5 px-m"
          : "text-base py-xs px-l"
      }`}
    >
      <span>{content}</span>
    </button>
  );
};

Button.propTypes = {
  size: PropTypes.oneOf(["sm", "md", "lg"]),
  disabled: PropTypes.bool,
  variant: PropTypes.oneOf(["primary", "secondary"]),
  className: PropTypes.string,
  content: PropTypes.string,
};
