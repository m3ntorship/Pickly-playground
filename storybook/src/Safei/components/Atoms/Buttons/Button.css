/* //using css for colors not available in config and used very few times. */
.primary:hover{
    background-color: #5F3DC4;
}
.primary:focus,.secondary:hover{
    border: 2px solid #5034A6;
}
.secondary:hover{
    background-color: #FAFAFC;
}
.text:hover{
    color:#0092C9;
}
.text:focus span{
    text-decoration: underline;
}