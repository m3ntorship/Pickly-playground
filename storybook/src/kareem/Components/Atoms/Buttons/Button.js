import React from "react";
import PropTypes from "prop-types";
import clsname from "classnames";

export const BUTTON_OPTIONS = {
  SIZE: {
    SMALL: "small",
    MEDIUM: "medium",
    LARGE: "large",
  },
};

export function Button({ primary, children, size }) {
  return (
    <div>
      <button
        type="button"
        className={clsname(
          primary
            ? "bg-primary text-white font-sans rounded-full focus:ring-2 focus:ring-primary-shd9 focus:outline-none disabled:opacity-25 mr -4"
            : "text-primary bg-white font-sans border border-primary rounded-full  focus:ring-2 focus:ring-primary-shd9 focus:outline-none mr-4 disabled:opacity-25",
          {
            "py-3.5 px-8 font-bold text-md": size === BUTTON_OPTIONS.SIZE.LARGE,
          },
          {
            "py-2 px-6 text-base font-medium":
              size === BUTTON_OPTIONS.SIZE.MEDIUM,
          },
          {
            "py-1.5 px-4 font-medium text-sm":
              size === BUTTON_OPTIONS.SIZE.SMALL,
          }
        )}
      >
        {children}
      </button>
    </div>
  );
}

Button.propTypes = {
  /**
   * Is this the principal call to action on the page?
   */
  primary: PropTypes.bool,
  /**
   * What background color to use
   */
  children: PropTypes.string,
  /**
   * How large should the button be?
   */
  size: PropTypes.oneOf(["small", "medium", "large"]),
};

Button.defaultProps = {
  primary: true,
  size: "medium",
  children: this.props.children,
};

export default Button;
