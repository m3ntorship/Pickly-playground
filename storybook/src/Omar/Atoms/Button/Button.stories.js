import React from "react";

import { Button } from "./Button";

export default {
  title: "Omar/Atoms/Button",
  component: Button,
};

const Template = (args) => <Button {...args} />;

export const Default = Template.bind({});

Default.args = {
  size: "sm",
  variant: "primary",
  disabled: false,
  children: "Button Text",
  leftIcon: false,
  rightIcon: true,
};
