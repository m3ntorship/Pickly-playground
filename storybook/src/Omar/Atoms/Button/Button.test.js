import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import { Button } from "./Button";

it("should click", () => {
  const mockClick = jest.fn();
  render(<Button onClick={mockClick}></Button>);
  const btn = screen.getByRole("button");
  userEvent.click(btn);
  expect(mockClick).toHaveBeenCalledTimes(1);
});

describe("Button Testing", () => {
  it("Button default testing with primary, size small and content 'button'", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Button disable state", () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Button with size medium", () => {
    const tree = renderer.create(<Button size="md" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Button with size large", () => {
    const tree = renderer.create(<Button size="lg" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Button with variant secondary", () => {
    const tree = renderer.create(<Button variant="secondary" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
