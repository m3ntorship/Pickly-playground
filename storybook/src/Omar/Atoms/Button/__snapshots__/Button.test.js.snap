// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Testing Button default testing with primary, size small and content 'button' 1`] = `
<button
  className="rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:outline-none focus:border-primary-shd9 text-sm py-1.5 px-m"
  data-testid="btn-case"
  disabled={false}
/>
`;

exports[`Button Testing Button disable state 1`] = `
<button
  className="rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:outline-none focus:border-primary-shd9 text-sm py-1.5 px-m"
  data-testid="btn-case"
  disabled={true}
/>
`;

exports[`Button Testing Button with size large 1`] = `
<button
  className="rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:outline-none focus:border-primary-shd9 text-md py-3.5 px-xl"
  data-testid="btn-case"
  disabled={false}
/>
`;

exports[`Button Testing Button with size medium 1`] = `
<button
  className="rounded-full border disabled:opacity-30 flex items-center bg-primary text-white hover:bg-primary-shd9 focus:outline-none focus:border-primary-shd9 text-base py-xs px-l"
  data-testid="btn-case"
  disabled={false}
/>
`;

exports[`Button Testing Button with variant secondary 1`] = `
<button
  className="rounded-full border disabled:opacity-30 flex items-center bg-white text-primary border-primary border-solid hover:bg-grey-bg2 focus:border-2 focus:outline-none text-sm py-1.5 px-m"
  data-testid="btn-case"
  disabled={false}
/>
`;
